frontend-wasm/src/code_translator/mod.rs

```diff
@@ -0,0 +1,1035 @@
+            let name = mod_info.global_name(global_index);
+            let ty = mod_info.globals[global_index].ty.clone();
+            state.push1(builder.ins().load_symbol(name, ty, span));
+        }
+        Operator::GlobalSet { global_index } => {
+            let global_index = GlobalIndex::from_u32(*global_index);
+            let name = mod_info.global_name(global_index);
+            let ty = (&mod_info.globals[global_index]).ty.clone();
+            let ptr = builder
+                .ins()
+                .symbol_addr(name, Ptr(ty.clone().into()), span);
+            let val = state.pop1();
+            builder.ins().store(ptr, val, span);
+        }
+        /********************************* Stack misc **************************************/
+        Operator::Drop => _ = state.pop1(),
+        Operator::Select => {
+            let (arg1, arg2, cond) = state.pop3();
+            // if cond is not 0, return arg1, else return arg2
+            // https://www.w3.org/TR/wasm-core-1/#-hrefsyntax-instr-parametricmathsfselect%E2%91%A0
+            let zero = builder.ins().i32(0, span);
+            let cond_i1 = builder.ins().neq(cond, zero, span);
+            state.push1(builder.ins().select(cond_i1, arg1, arg2, span));
+        }
+        Operator::Unreachable => {
+            builder.ins().unreachable(span);
+            state.reachable = false;
+        }
+        Operator::Nop => {}
+        /***************************** Control flow blocks *********************************/
+        Operator::Block { blockty } => translate_block(blockty, builder, state, mod_info, span)?,
+        Operator::Loop { blockty } => translate_loop(blockty, builder, state, mod_info, span)?,
+        Operator::If { blockty } => translate_if(blockty, state, builder, mod_info, span)?,
+        Operator::Else => translate_else(state, builder, span)?,
+        Operator::End => translate_end(state, builder, span),
+
+        /**************************** Branch instructions *********************************/
+        Operator::Br { relative_depth } => translate_br(state, relative_depth, builder, span),
+        Operator::BrIf { relative_depth } => translate_br_if(*relative_depth, builder, state, span),
+        Operator::BrTable { targets } => translate_br_table(targets, state, builder, span)?,
+        Operator::Return => translate_return(state, builder, diagnostics, span)?,
+        /************************************ Calls ****************************************/
+        Operator::Call { function_index } => {
+            translate_call(state, builder, function_index, mod_info, span, diagnostics)?;
+        }
+        /******************************* Memory management *********************************/
+        Operator::MemoryGrow { .. } => {
+            let arg = state.pop1();
+            state.push1(builder.ins().mem_grow(arg, span));
+        }
+        Operator::MemorySize { .. } => {
+            // Return total Miden memory size
```

Comment:
We should add an instruction for this, similar to `mem_grow`, otherwise it won't match what is returned by `mem_grow`, and that may cause issues. Is `memory_size` actually used anywhere that we've seen yet? IIRC `wee_alloc` and `dlmalloc` both just use `mem_grow`, but I could be misremembering.

Reply:
I agree. There is no `memory_size` in `dlmalloc` code. I have not checked `wee_alloc` yet.